protocols/index.md

```diff
@@ -0,0 +1,33 @@
+# CloudState Documentation and Specification format rules
+
+The CloudState Protocol Specification MUST use [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt) definitions.
+
+Use the following formatting and Markdown syntax to define documentation and specification of the CloudState Protocol.
+The template/example below is proven to work well with the `protoc-gen-doc` plugin, so please stick to it 100%.
+
+```
+//The *EntityDiscovery* Service is the entrypoint for CloudState Proxies
+//
+//---
+//
+//`Specification`:
+//
+//  - A `User Function` MUST implement EntityDiscovery
+//  - `discover`
+//    - *MUST* be invoked with a *valid* `ProxyInfo`
+//    - *MUST* be *idempotent*
+//    - *MAY* be called any number of times by its CloudState Proxy
+//    - *MUST* yield a failure if the provided `ProxyInfo` is not deemed compatible with the User Function
+//  - `reportError`
+//    - *MUST* only be invoked to inform a User Function about violations of the CloudState Specification
```

Comment:
I think we can use the reportError return to report a version mismatch.

Reply:
After reviewing some parts of the code I noticed that the only way to return an incompatibility error today would be to throw an exception. Forget what I said about reportError as it obviously wouldn't work.
We may have to add more information in ServiceInfo